### PR TITLE
fix/popup

### PR DESCRIPTION
**Pull request by mnicpt**
- Ensure unload event listener that closes popup when parent closes is removed by passing in a named function
- Ensure multiple popups cannot occur by closing the old one and opening a new one if the button is tapped again when cancelling out of native checkout flow by multi-tasking

https://engineering.paypalcorp.com/jira/browse/DTCHKINT-354

### CODE REVIEW

File: src/payment-flows/native.js [hunk 1]
```diff
@@ -248,7 +248,12 @@ function initNative({ props, components, config, payment, serviceData } : InitOp
         throw new Error(`Can not run native flow without firebase config`);
     }
 
-    const clean = cleanup();
+    if (window.clean) {
```
Comment (bluepnume) [inline src/payment-flows/native.js:251]: Can we use a module-level global at the top of `native.js`, i.e. just a `let clean;`

File: src/payment-flows/native.js [hunk 2]
```diff
@@ -496,9 +503,11 @@ function initNative({ props, components, config, payment, serviceData } : InitOp
         const nativeUrl = getNativeUrlForAndroid({ sessionUID });
 
         const nativeWin = popup(nativeUrl);
-        window.addEventListener('unload', () => {
+
+        const closePopup = () => {
             nativeWin.close();
-        });
+        };
+        window.addEventListener('pagehide', closePopup);
```
Comment (bluepnume) [inline src/payment-flows/native.js:510]: Interesting. Why `pagehide` vs `unload`?

Comment (mnicpt) [reply src/payment-flows/native.js:510]: I found that pagehide was fired more consistently than unload when redirecting.

**Comment by gregjopa** [comment]
I pulled this down locally to test. I noticed that there's one failing test:

```
TOTAL: 1 FAILED, 103 SUCCESS


1) should render a button with createOrder, click the button, and render checkout via popup to native path in Chrome with multiple clicks
     native chrome cases
     Error: Expected window to not close
```

There's a version of this test for both Android/Chrome and iOS. Only the test using the android user agent string seems to fail. How do we best resolve this failing test?

**Comment by mnicpt** [comment]
@gregjopa Yeah, I'm still working on that.  It's expecting the popup to not be closed.

**Comment by mnicpt** [comment]
@bluepnume @gregjopa I have fixed the tests to comply with the new changes.

**Review by bluepnume — APPROVED** [review]

**Review by gregjopa — APPROVED** [review]
Looks good to me! I tested the pagehide event on my phone using this site and it looks good: https://event-logger.glitch.me/

**Pull request merged**